packages/password/src/index.mts

```diff
@@ -1,39 +1,110 @@
-import type { Prompt } from '@inquirer/type';
-import input from '@inquirer/input';
+import {
+  createPrompt,
+  useState,
+  useKeypress,
+  usePrefix,
+  isEnterKey,
+  type PromptConfig,
+} from '@inquirer/core';
 import chalk from 'chalk';
 import ansiEscapes from 'ansi-escapes';
+import { Context } from '@inquirer/type';
 
-type InputConfig = Parameters<typeof input>[0];
-type PasswordConfig = Omit<InputConfig, 'transformer' | 'default'> & {
+type PasswordConfig = PromptConfig<{
   mask?: boolean | string;
+  validate?: (value: string) => boolean | string | Promise<string | boolean>;
+  allowShowPassword?: boolean;
+}>;
+
+const transformer = (
+  str: string,
+  mask: string | boolean | undefined,
+  { isFinal }: { isFinal: boolean },
+) => {
+  if (mask) {
+    const maskChar = typeof mask === 'string' ? mask : '*';
+    return maskChar.repeat(str.length);
+  }
+
+  if (!isFinal) {
+    return `${chalk.dim('[input is masked]')}${ansiEscapes.cursorHide}`;
+  }
+
+  return '';
 };
 
-const password: Prompt<string, PasswordConfig> = (config, context) => {
+export default (config: PasswordConfig, context?: Context | undefined) => {
   if ('transformer' in config) {
```

Comment:
By moving the `if('transformer' in config) ...` inside the `createPrompt` the test which checks that it will throw if the `transformer` property is provided fails

Reply:
Any idea on why it could fail?

Reply:
Had to change the test slightly